clang/test/SemaSYCL/kernel-arg-opt-report.cpp

```diff
@@ -0,0 +1,42 @@
+// RUN: %clang_cc1 -triple spir64-unknown-unknown-sycldevice -fsycl-is-device \
+// RUN: -Wno-sycl-2017-compat -emit-llvm-bc %s -o %t-host.bc -opt-record-file %t-host.yaml
```

Comment:
Could you expand this test to check for basic things in the opt-report file?  I would like to see that to understand the changes in CodeGenFunction better.